### DISCUSSION TITLE
long-running server executing search requests against qdrant database: should it keep the client connection open, or create a new client and then close it after serving each request?

### DISCUSSION THREAD

**Discussion by maspotts** [qdrant/qdrant-client#768] [Q&A]
I've created a chatbot server, which uses a `QdrantClient` instance to execute searches against the database (via a langchain `QdrantVectorStore` instance): should I keep the client instance connection open as long as my server is running?  Or should I create a new client instance each time I receive an HTTP request and then close it again after it's served the request?  (And if the latter, it would be great if there was a `QdrantClient.reopen()` method, because the `QdrantVectorStore` class doesn't provide a way to update the client if the old client instance is closed and a new client instance is created!).

**Comment by octocat20** [comment]
Keep one `QdrantClient` open for the lifetime of the server and reuse it across requests.

Creating a new client per HTTP request just adds overhead. Same idea if you're using LangChain's `QdrantVectorStore`: build it once with that shared client and keep it around.

Only create a new client if the URL or API key changes, or in tests where you want isolation.
